core-rust/state-manager/src/staging/result.rs

```diff
@@ -0,0 +1,360 @@
+
+lazy_static! {
+    static ref EMPTY_STATE_DIFF: StateDiff = StateDiff::new();
+}
+
+pub struct ProcessedResult {
+    /// Raw transaction receipt.
+    pub transaction_receipt: TransactionReceipt,
+    /// The processing results, applicable only for committed transactions (i.e. not rejected).
+    pub processed_commit: Option<ProcessedTransactionCommit>,
+}
+
+impl ProcessedResult {
+    pub fn from_processed<S: ReadableLayeredTreeStore + ReadableTxAndRxTreeStore>(
+        store: &S,
+        epoch_transaction_identifiers: &EpochTransactionIdentifiers,
+        parent_transaction_identifiers: &CommittedTransactionIdentifiers,
+        transaction_accumulator_hash: AccumulatorHash,
+        transaction_hash: &LedgerPayloadHash,
+        transaction_receipt: TransactionReceipt,
+    ) -> ProcessedResult {
+        let ledger_receipt = LedgerTransactionReceipt::try_from(transaction_receipt.clone()).ok();
+        let processed_commit = ledger_receipt.map(|ledger_receipt| {
+            let state_diff = Self::extract_state_diff(&transaction_receipt);
+            let state_tree_diff = Self::compute_state_tree_update(
+                store,
+                parent_transaction_identifiers.state_version,
+                state_diff,
+            );
+            let transaction_tree_diff = Self::compute_accu_tree_update::<S, TransactionTreeHash>(
+                store,
+                epoch_transaction_identifiers.state_version,
+                &epoch_transaction_identifiers.transaction_hash,
+                parent_transaction_identifiers.state_version,
+                TransactionTreeHash::from_raw_bytes((*transaction_hash).into_bytes()),
+            );
+            let receipt_tree_diff = Self::compute_accu_tree_update::<S, ReceiptTreeHash>(
+                store,
+                epoch_transaction_identifiers.state_version,
+                &epoch_transaction_identifiers.receipt_hash,
+                parent_transaction_identifiers.state_version,
+                ReceiptTreeHash::from_raw_bytes(ledger_receipt.get_hash().into_bytes()),
+            );
+            let ledger_hashes = LedgerHashes {
+                state_root: state_tree_diff.new_root,
+                transaction_root: *transaction_tree_diff.slice.root(),
+                receipt_root: *receipt_tree_diff.slice.root(),
+            };
+            ProcessedTransactionCommit {
+                transaction_accumulator_hash,
+                ledger_hashes,
+                state_tree_diff,
+                transaction_tree_diff,
+                receipt_tree_diff,
+            }
+        });
+        Self {
+            transaction_receipt,
+            processed_commit,
+        }
+    }
+
+    pub fn receipt(&self) -> &TransactionReceipt {
+        &self.transaction_receipt
+    }
+
+    pub fn state_diff(&self) -> &StateDiff {
+        Self::extract_state_diff(&self.transaction_receipt)
+    }
+
+    pub fn commit(&self) -> &ProcessedTransactionCommit {
```

Comment:
`commit` sounds like a verb to me. Perhaps `commit_details` or `get_commit_details` is clearer to me?